docs/4.0-migration-guide/pake.md

```diff
@@ -1,5 +1,111 @@
 
-TODO
+In Mbed TLS 3.6, the PAKE interface implemented version [1.1 Beta](https://arm-software.github.io/psa-api/crypto/1.1/ext-pake/) of the PAKE extension. There has been a number of [changes between the beta and the final version](https://arm-software.github.io/psa-api/crypto/1.2/ext-pake/appendix/history.html#changes-between-beta-1-and-final) of the API. The changes that require applications to update their code are detailed in the following subsections.
+
+Note that TF-PSA-Crypto 1.0 still only implements `PSA_ALG_JPAKE` (and only on elliptic curves, specifically only on secp256r1). Support for SPAKE2+ is likely to be added in a future version but is not there yet.
+
+### Combine `psa_pake_set_password_key()` with `psa_pake_setup()`
+
+The function `psa_pake_set_password_key()` has been removed. Its `key` argument is now passed to `psa_pake_setup()` which has gained a new `key` parameter.
+
+Before:
+
+```
+status = psa_pake_setup(&operation, &cipher_suite);
+if (status != PSA_SUCCESS) // error handling ommited for brevity
+status = psa_pake_set_password_key(&operation, key);
+if (status != PSA_SUCCESS) // error handling ommited for brevity
+```
+
+Now:
+
+```
+status = psa_pake_setup(&operation, key, &cipher_suite);
+if (status != PSA_SUCCESS) // error handling ommited for brevity
+```
+
+### Move the hash algorithm parameter into the algorithm identifier
+
+The function `psa_pake_cs_set_hash()` has been removed. Its `hash` argument is now passed to `PSA_ALG_JPAKE()` which is now a function-like macro with one parameter.
+
+Before:
+
+```
+psa_pake_cs_set_algorithm(&cipher_suite, PSA_ALG_JPAKE);
+psa_pake_cs_set_hash(&cipher_suite, PSA_ALG_SHA_256);
+```
+
+Now:
+
+```
+psa_pake_cs_set_algorithm(&cipher_suite, PSA_ALG_JPAKE(PSA_ALG_SHA_256));
+```
+
+To check if a given algorithm is J-PAKE, the new `PSA_ALG_IS_JPAKE()` macro has been added.
+
+Before: `if (alg == PSA_ALG_JPAKE)`
+
+Now: `if (PSA_ALG_IS_JPAKE(alg))`
+
+The function `psa_pake_cs_get_hash()` has also been removed.
+
+### Replace `psa_pake_get_implicit_key()` with `psa_pake_get_shared_key()`
+
+The function `psa_pake_get_implicit_key()`, which injects the shared secret into a key derivation operation, has been removed. Its replacement is `psa_pake_get_shared_key()` which stores the shared secret in a new key. That new key can then be used as part of a key derivation operation.
+
+Before:
+
+```
+status = psa_pake_get_implicit_key(&pake_op, &derivation_op);
+if (status != PSA_SUCCESS) // error handling ommited for brevity
+```
+
+Now:
+
+```
+psa_algorithm_t alg = PSA_ALG_TLS12_PSK_TO_MS(PSA_ALG_SHA_256);
+                      // or other key derivation algorithm
```

Comment:
It has to be the algorithm that `derivation_op` was set up for.

I think it would be helpful to include the code to set up and use the derivation operation, in both the before and after snippets.

A typical audience for the migration guide is someone who may be upgrading multiple libraries used in a project, and who isn't very familiar with the code that they're upgrading. So we should guide them with respect to understanding their existing application code.